paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethodKtx.kt

```diff
@@ -159,17 +160,25 @@ private fun checkPaymentIntentRequirements(
     requirements: Set<PIRequirement>?,
     paymentIntent: PaymentIntent,
     config: PaymentSheet.Configuration?
-) =
-    requirements?.map { requirement ->
+): Boolean {
+    return requirements?.map { requirement ->
         when (requirement) {
-            Delayed -> config?.allowsDelayedPaymentMethods == true
-            ShippingAddress ->
-                paymentIntent.shipping?.name != null &&
-                    paymentIntent.shipping?.address?.line1 != null &&
-                    paymentIntent.shipping?.address?.country != null &&
-                    paymentIntent.shipping?.address?.postalCode != null
+            Delayed -> {
+                config?.allowsDelayedPaymentMethods == true
+            }
+            ShippingAddress -> {
+                val forceAllow = config?.allowsPaymentMethodsRequiringShippingAddress == true
```

Comment:
nit: not a big fan of the name here. Can we keep it as `allowsPaymentMethodsRequiringShippingAddress`

Reply:
I think it’s a fairly accurate naming: The caller is forcing the payment method to be shown even though it shouldn’t be (based on the intent).